tests/test_functional.py

```diff
@@ -53,6 +54,21 @@ class Opt:
 
         self.assertRaises(TypeError, parse_args, Opt, [])
 
+    def test_add_dataclass_options(self):
+        @dataclass
+        class Opt:
+            x: int = 42
+            y: bool = False
+        argpument_parser = argparse.ArgumentParser()
+        add_dataclass_options(Opt, argpument_parser)
+        params = argpument_parser.parse_args([])
+        print(params)
+        self.assertEqual(42, params.x)
+        self.assertEqual(False, params.y)
```

Comment:
`parse_args` returns a `namespace` object not a `Opt` so it is not expected to have the defaults set.